simplq/src/store/queues/queuesSlice.js

```diff
@@ -0,0 +1,28 @@
+/* eslint-disable no-param-reassign */
+
+import { createSlice } from '@reduxjs/toolkit';
+import { fetchQueues, deleteQueue } from './queuesAsyncActions';
+
+const queuesSlice = createSlice({
+  name: 'queues',
+  initialState: [],
```

Comment:
There are three types of information that we would store for queues:

1) List of queues, this is what you have implemented here. (yeah, `UserQueues` looks like a better name)
2) Status of a single queue 
3) Detailed queue object (used in admin page)

Do we plan to have three different slices for these? What is the best approach?

Reply:
After a quick look at data:
```js
const state = {
  queues: [],
  selectedQueue: {}, // It can be Admin queue, Join page queue, or status page Queue
  queueStatus: {}, //Updated when selectedQueue is updated
  tokens: [] // Updated with Admin page queue
}
```

Reply:
we could expand on any of the above slices. Rearrange, extract, normalize...

```js
const state = {
  queues: {
    loading: false,
    ids: [],
    byId: {}
  },
  selectedQueue: {},
  queueStatus: {}, 
  tokens: {
    ids: [],
    byId: {}
  }
}
```

Reply:
Interesting, why store `ids: []`? Isn't it already covered by `byId: {}`?

Reply:
At this stage, it doesn't really matter. For more complex future [normalizing state](https://redux.js.org/recipes/structuring-reducers/normalizing-state-shape) and [connecting more components](https://redux.js.org/recipes/structuring-reducers/normalizing-state-shape) is recommeded.